src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -124,30 +135,111 @@ object Instance {
       * @param scheduledInstance instance in a Scheduled state
       * @return new instance in a provisioned state
       */
-    def apply(scheduledInstance: Instance, agentInfo: AgentInfo, networkInfo: core.task.state.NetworkInfo, app: AppDefinition, now: Timestamp): Instance = {
+    def apply(
+      scheduledInstance: Instance,
+      agentInfo: AgentInfo,
+      networkInfo: core.task.state.NetworkInfo,
+      app: AppDefinition,
+      now: Timestamp): Instance = {
       require(scheduledInstance.isScheduled, s"Instance '${scheduledInstance.instanceId}' must not be in state '${scheduledInstance.state.condition}'. Scheduled instance is required to create provisioned instance.")
 
-      val task = Task(
-        taskId = Task.Id.forInstanceId(scheduledInstance.instanceId, None),
-        runSpecVersion = app.version,
-        status = Task.Status(
-          stagedAt = now,
-          condition = Condition.Created,
-          networkInfo = networkInfo
-        )
+      val taskIds: Seq[Task.Id] = if (scheduledInstance.hasReservation) {
+        val originalIds = if (scheduledInstance.tasksMap.nonEmpty) {
+          scheduledInstance.tasksMap.keys
+        } else {
+          Seq(Task.Id.forInstanceId(scheduledInstance.instanceId, None))
+        }
+        originalIds.map(ti => Task.Id.forResidentTask(ti)).to[Seq]
```

Comment:
I'd be fine with `originalIds.map(Task.Id.forResidentTask)`. I've talked to Alex yesterday. He agreed that we probably want to have a different reservation id scheme. Currently the reservation id is the same as the task id, right?

Reply:
yes